pkg/core/blockchain.go

```diff
@@ -325,8 +332,16 @@ func (bc *Blockchain) init() error {
 	if ver.Value != version {
 		return fmt.Errorf("storage version mismatch betweeen %s and %s", version, ver.Value)
 	}
-	bc.dao.StoragePrefix = ver.Prefix
-	bc.persistent.StoragePrefix = ver.Prefix // not strictly needed but we better be consistent here
+	if ver.StateRootInHeader != bc.config.StateRootInHeader {
+		return fmt.Errorf("StateRootInHeader setting mismatch (config=%t, db=%t)",
+			ver.StateRootInHeader, bc.config.StateRootInHeader)
+	}
+	if ver.P2PSigExtensions != bc.config.P2PSigExtensions {
+		return fmt.Errorf("P2PSigExtensions setting mismatch (old=%t, new=%t",
+			ver.P2PSigExtensions, bc.config.StateRootInHeader)
```

Comment:
The second parameter is invalid, it should be `bc.config.P2PSigExtensions`

Reply:
Fixed.